File: src/main/java/me/sullivan/glasslang/parser/nodes/ListNode.java
package me.sullivan.glasslang.parser.nodes;

import me.sullivan.glasslang.interpreter.Interpreter;
import me.sullivan.glasslang.interpreter.primitives.types.Primitive;

import java.util.List;

public class ListNode extends Node
{
    private final List<Node> nodes;

    public ListNode(List<Node> nodes)
    {
        super(null);
        this.nodes = nodes;
    }

    @Override
    public List<Node> getValue()
    {
        return nodes;
    }

    @Override
    public String toString()
    {
        return "List" + nodes;
    }

    @Override
    public Primitive<?> visitor(Interpreter interpreter)
    {
        return interpreter.visitListNode(this);
    }
}

File: src/me/sullivan/glasslang/parser/nodes/ListNode.kt
package me.sullivan.glasslang.parser.nodes

import me.sullivan.glasslang.interpreter.Interpreter
import me.sullivan.glasslang.interpreter.primitives.Primitive

class ListNode(private var list: List<Node>) : Node(null) {

    override fun toString(): String
    {
        return "List[$list]"
    }

    override fun getValue(): List<Node>
    {
        return list
    }

    override fun visitor(interpreter: Interpreter): Primitive<*>
    {
        return interpreter.visitListNode(this)
    }

}